embedded-tests/src/test/java/org/apache/druid/testing/embedded/msq/EmbeddedMSQRealtimeQueryTest.java

```diff
@@ -0,0 +1,383 @@
+           .addServer(router);
+
+    return cluster;
+  }
+
+  @BeforeEach
+  void setUpEach()
+  {
+    msqApis = new EmbeddedMSQApis(cluster, overlord);
+    topic = dataSource = EmbeddedClusterApis.createTestDatasourceName();
+
+    // Create Kafka topic.
+    kafka.createTopicWithPartitions(topic, 2);
+
+    // Submit a supervisor.
+    final KafkaSupervisorSpec kafkaSupervisorSpec = createKafkaSupervisor();
+    final Map<String, String> startSupervisorResult =
+        cluster.callApi().onLeaderOverlord(o -> o.postSupervisor(kafkaSupervisorSpec));
+    Assertions.assertEquals(Map.of("id", dataSource), startSupervisorResult);
+
+    // Send data to Kafka.
+    final QueryableIndexCursorFactory wikiCursorFactory =
+        new QueryableIndexCursorFactory(TestIndex.getMMappedWikipediaIndex());
+    final RowSignature wikiSignature = wikiCursorFactory.getRowSignature();
+    kafka.produceRecordsToTopic(
+        FrameTestUtil.readRowsFromCursorFactory(wikiCursorFactory)
+                     .map(row -> {
+                       final Map<String, Object> rowMap = new LinkedHashMap<>();
+                       for (int i = 0; i < row.size(); i++) {
+                         rowMap.put(wikiSignature.getColumnName(i), row.get(i));
+                       }
+                       try {
+                         return new ProducerRecord<>(
+                             topic,
+                             ByteArrays.EMPTY_ARRAY,
+                             TestHelper.JSON_MAPPER.writeValueAsBytes(rowMap)
+                         );
+                       }
+                       catch (JsonProcessingException e) {
+                         throw new RuntimeException(e);
+                       }
+                     })
+                     .toList()
+    );
+
+    // Wait for it to be loaded.
+    indexer.latchableEmitter().waitForEventAggregate(
+        event -> event.hasMetricName("ingest/events/processed")
+                      .hasDimension(DruidMetrics.DATASOURCE, Collections.singletonList(dataSource)),
+        agg -> agg.hasSumAtLeast(totalRows)
+    );
+  }
+
+  @AfterEach
+  void tearDownEach() throws ExecutionException, InterruptedException, IOException
+  {
+    final Map<String, String> terminateSupervisorResult =
+        cluster.callApi().onLeaderOverlord(o -> o.terminateSupervisor(dataSource));
+    Assertions.assertEquals(Map.of("id", dataSource), terminateSupervisorResult);
+
+    // Cancel all running tasks, so we don't need to wait for them to hand off their segments.
+    try (final CloseableIterator<TaskStatusPlus> it = cluster.leaderOverlord().taskStatuses(null, null, null).get()) {
+      while (it.hasNext()) {
+        cluster.leaderOverlord().cancelTask(it.next().getId());
+      }
+    }
+
+    kafka.deleteTopic(topic);
+  }
+
+  @Test
+  @Timeout(60)
+  public void test_selectCount_task_default()
+  {
+    final String sql = StringUtils.format("SELECT COUNT(*) FROM \"%s\"", dataSource);
```

Comment:
We probably don't need to use `StringUtils.format()` since `msqApis.runTaskSql()` already accepts formatting arguments.